builder/llpcBuilderImplMisc.cpp

```diff
@@ -100,6 +100,27 @@ Instruction* BuilderImplMisc::CreateKill(
     return CreateIntrinsic(Intrinsic::amdgcn_kill, {}, getFalse(), nullptr, instName);
 }
 
+// =====================================================================================================================
+// Create a demote to helper invocation operation. Only allowed in a fragment shader.
+Instruction* BuilderImplMisc::CreateDemoteToHelperInvocation(
+    const Twine& instName) // [in] Name to give instruction(s)
+{
+    // Treat a demote as a kill for the purposes of disabling middle-end optimizations.
+    auto pResUsage = getContext().GetShaderResourceUsage(ShaderStageFragment);
+    pResUsage->builtInUsage.fs.discard = true;
+
+    return CreateIntrinsic(Intrinsic::amdgcn_wqm_demote, {}, getFalse(), nullptr, instName);
+}
+
+// =====================================================================================================================
+// Create a helper invocation query. Only allowed in a fragment shader.
+Value* BuilderImplMisc::CreateIsHelperInvocation(
+    const Twine& instName) // [in] Name to give instruction(s)
+{
+    auto pIsHelper = CreateIntrinsic(Intrinsic::amdgcn_wqm_helper, {}, {}, nullptr, instName);
```

Comment:
What is the return value of amdgcn_wqm_helper? pIsHelper means this is a helper lane? So why we still need a negation operation by CreateNot?

Reply:
wqm_helper is actually an alternate version of ps_live.  I agree the choice of name is misleading, it should probably be wqm_live, I'll change this in the upstream copy.

Reply:
OK. Then change to this: pIsHelper -> pIsNotHelper